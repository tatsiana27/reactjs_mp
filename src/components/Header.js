import React from 'react';
import PropTypes from 'prop-types';

import { CommonComponent } from './CommonComponent';
import './header.scss';

export function Header({ children }) {
    return (
        <div className="header">
            <CommonComponent/>
            <div>{children}</div>
        </div>
    );
}

Header.defaultProps = {
    children: null,
};

Header.propTypes = {
    children: PropTypes.node.isRequired,
};