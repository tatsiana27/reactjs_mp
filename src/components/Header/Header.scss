@import '../../styles/variables';

.header {
  position: relative;
  background-size: cover;
  background-position: 50%;
  background-image: url("../../static/images/netflix-bg.png");
  padding: 40px;
  color: white;
  .common_title {
    padding: {
      top: 0;
      bottom: 0;
    };
    background-color: transparent;
  }
}
