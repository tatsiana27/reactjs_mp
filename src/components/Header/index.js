import React from 'react';
import PropTypes from 'prop-types';

import { CommonComponent } from '../CommonComponent';
import './Header.scss';

export function Header({ children }) {
  return (
    <div className="header">
      <CommonComponent/>
      <div>{children}</div>
      <div className="overlay"></div>
    </div>
  );
}

Header.defaultProps = {
  children: null,
};

Header.propTypes = {
  children: PropTypes.node,
};
