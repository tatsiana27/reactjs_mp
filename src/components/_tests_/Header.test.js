import React from 'react';
import { Header } from '../Header';
import { shallow } from 'enzyme';

describe('<Header />', () => {
    test('should render Header Component with children', () => {
        const wrapper = shallow(
            <Header>
                <div>mocks children</div>
            </Header>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render Header Component without children', () => {
        const wrapper = shallow(
            <Header />
        );

        expect(wrapper).toMatchSnapshot();
    });
});