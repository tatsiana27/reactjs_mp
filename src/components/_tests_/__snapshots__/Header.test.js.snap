// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header Component with children 1`] = `
<div
  className="header"
>
  <CommonComponent />
  <div>
    <div>
      mocks children
    </div>
  </div>
</div>
`;

exports[`<Header /> should render Header Component without children 1`] = `
<div
  className="header"
>
  <CommonComponent />
  <div />
</div>
`;
